python/graphstorm/gconstruct/transform.py

```diff
@@ -133,31 +181,19 @@ def process_features(data, ops):
     ----------
     data : dict
         The data stored as a dict.
-    ops : list of tuples
-        The operations. Each tuple contains two elements. The first element
-        is the data name and the second element is a Python function
-        to process the data.
+    ops : list of FeatTransform
+        The operations that transform features.
 
     Returns
     -------
     dict : the key is the data name, the value is the processed data.
     """
     new_data = {}
-    for feat_col, feat_name, dtype, op in ops:
-        # If the transformation is defined on the feature.
-        if op is not None:
-            res = op(data[feat_col])
-            if isinstance(res, dict):
-                for key, val in res.items():
-                    new_data[key] = val
-            else:
-                new_data[feat_name] = res
-        # If the required data type is defined on the feature.
-        elif dtype is not None:
-            new_data[feat_name] = data[feat_col].astype(dtype)
-        # If no transformation is defined for the feature.
-        else:
-            new_data[feat_name] = data[feat_col]
+    for op in ops:
+        res = op(data[op.col_name])
+        assert isinstance(res, dict)
+        for key, val in res.items():
+            new_data[key] = val
     return new_data
 
 def get_valid_label_index(label):
```

Comment:
Can we treat negative integer, i.e., -1 as invalid labels for integer label data?

Reply:
currently, we only treat NaN as invalid label values.